new/resources-anywhere.md

```diff
@@ -0,0 +1,211 @@
+---
+RFC: unassigned
+Author: John Keiser <john@johnkeiser.com>
+Status: Draft
+Type: Standards Track
+---
+
+# Allow Resources In Any Namespace, Remove Method Missing
+
+This proposal addresses:
+- Allowing users to place Chef resources and providers anywhere in their tree, not just `Chef::Resource` and `Chef::Provider`
+- Eliminating the issue where resource and provider code that tries to access `File.x` actually get the File resource, having to type `::File` instead to get what they want.
+- Eliminating `method_missing` as a way to look up resources, in favor of explicit DSL (which gets rid of a great deal of confusing error message).
+
+## Motivation
+
+    As a <<user_profile>>,
+    I want to <<functionality>>,
+    so that <<benefit>>.
+
+## Specification
+
+### Resources outside of `Chef::Resource`
+
+Users who wish to register a resource named `my_resource` can presently do it by creating a class inside the `Chef::Resource` namespace and another in `Chef::Provider`.  Chef then uses magical namespace lookups to
```

Comment:
The provider resolver and resource resolver in Chef 12 actually already accomplish this.  There's still a fallback to magic for stuff that hasn't been fully converted yet, but if that was done then the magical namespace lookups would all go away, along with ripping out the old platform mapping.  Classes still have to inherit from Chef::Resource or Chef::Provider, but they can be named anything.